flutter-studio/src/io/flutter/module/FlutterModuleImporter.java

```diff
@@ -0,0 +1,149 @@
+    if (settingsFile == null) {
+      showHowToEditDialog();
+      return;
+    }
+
+    VirtualFile appDir = projectRoot.findChild("app");
+    if (appDir == null) {
+      showHowToEditDialog();
+      return;
+    }
+    VirtualFile buildFile = appDir.findChild("build.gradle");
+    if (buildFile == null) {
+      showHowToEditDialog();
+      return;
+    }
+
+    writer = new StringWriter();
+    StringWriter settingsWriter, buildWriter;
+    try {
+      settingsWriter = editFileContent(settingsFile, this::writeSettingsLineWithEdits);
+      buildWriter = editFileContent(buildFile, this::writeBuildLineWithEdits);
+    }
+    catch (IOException e) {
+      showHowToEditDialog();
+      return;
+    }
+
+    // After both buffers are full, write them both to their original files.
+    ApplicationManager.getApplication().runWriteAction(() -> {
+      try {
+        settingsFile.setBinaryContent(settingsWriter.toString().getBytes(Charset.defaultCharset()));
+        buildFile.setBinaryContent(buildWriter.toString().getBytes(Charset.defaultCharset()));
+        // Then queue Gradle sync to update everything. Do it here to ensure proper sequencing.
+      }
+      catch (IOException e) {
+        // Should not happen
+      }
+    });
+  }
+
+  private StringWriter editFileContent(VirtualFile file, Consumer<String> editFunction) throws IOException {
+    writer = new StringWriter();
+    StringWriter buildWriter = writer;
+    try (BufferedReader r = Files.newBufferedReader(Paths.get(file.getPath()), Charset.defaultCharset())) {
+      hasFinishedEditing = false;
+      r.lines().forEach(editFunction);
+      writer.close();
+    }
+    return writer;
+  }
+
+  private void writeSettingsLineWithEdits(String line) {
+    writeLineWithEdits(line, "include", this::writeSettingsModToBuffer);
+  }
+
+  private void writeBuildLineWithEdits(String line) {
+    writeLineWithEdits(line, "dependencies", this::writeBuildModToBuffer);
+  }
+
+  private void writeLineWithEdits(String line, String trigger, Supplier editFunction) {
+    if (hasFinishedEditing) {
+      writeLineToBuffer(line);
+    }
+    else {
+      if (line.trim().startsWith(trigger)) {
+        writeLineToBuffer(line);
+        hasFinishedEditing = (Boolean)editFunction.get();
+      }
+      else {
+        writeLineToBuffer(line);
+      }
+    }
+  }
+
+  private boolean writeSettingsModToBuffer() {
+    writeLineToBuffer("setBinding(new Binding([gradle: this]))");
+    writeLineToBuffer("evaluate(new File(");
+    writeLineToBuffer("  settingsDir.parentFile,");
+    writeLineToBuffer("  '" + myRelativePath + "'");
+    writeLineToBuffer("))");
+    return true;
+  }
+
+  private boolean writeBuildModToBuffer() {
+    writeLineToBuffer("    implementation project(':flutter')");
+    return true;
+  }
+
+  private void writeLineToBuffer(String line) {
+    writer.write(line);
+    writer.write(myNewline);
+  }
+
+  private static void showHowToEditDialog() {
```

Comment:
Is this meant to be blank?

Reply:
No, it should show something to help the user make manual edits. I'm thinking I'll just keep it short and include a link to the add2app doc on github. That work for you? (I had been thinking of keeping it self-contained and including more detail but I think that would be rather annoying/distracting if a large dialog keeps popping up.)

Here's a possibility:
>The Gradle files could not be updated automatically. To use your Flutter module in this Android app you'll need to edit the files as described in the [add2app documentation](https://github.com/flutter/flutter/wiki/Add-Flutter-to-existing-apps#make-the-host-app-depend-on-the-flutter-module).

Reply:
👍, sounds great re: keeping it simple and pointing people to the online docs